lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +469,53 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+
+    def get_chunk(self, trans, dataset, offset=0, ck_size=None):
+        bamfile = pysam.AlignmentFile(dataset.file_name, "rb")
+        ck_size = 100 # 100 lines
+        ck_data=""
+        lineNumber=0
+        if(offset == 0):
+            ck_data = bamfile.text
+        for f in bamfile.fetch(until_eof=True):
+            lineNumber+=1
```

Comment:
Since it's probably better to use the offset you can do `for line_number, f in enumerate(bamfile):`.
Also f is frequently use to indicate a file handle, in this situation it would probably be better to rename `f` to `read`, `alignment` or `segment`.